code/client/makepkginfo

```diff
@@ -385,6 +385,15 @@ def check_mode(option, opt, value, parser):
                                                             (opt, mode))
     setattr(parser.values, option.dest, value)
 
+def has_valid_install_critieria(catinfo):
+    """
+    Inspects compiled catalog info dictionary for valid install criteria.
+    Returns boolean.
+    """
+    results = []
+    for key in ['installs', 'receipts', 'installcheck_script']:
+        results.append(catinfo.has_key(key) and len(catinfo[key]) != 0)
+    any(results)
```

Comment:
Combining your suggestion DRYs it out a tad. What about this...?
```Python
def has_valid_install_critieria(catinfo):
    """
    Inspects compiled catalog info dictionary for valid install criteria.
    Returns boolean.
    """
    results = []
    keys = ['installs', 'receipts', 'installcheck_script', 'PayloadIdentifier']
    for key in keys:
        results.append(catinfo.get(key))
    return any(results)
```

And...

```Python
    if not has_valid_install_critieria(catinfo):
        item = catinfo['name']
        print >> sys.stderr, (
            "WARNING: %s contains no discernible install criteria!\n"
            "If you do not supply a receipt, install item, "
            "installcheck_script \n"
            "or PayloadIdentifier munki will have no way to determine \n"
            "the package's installation state. This can result in \n"
            "undesirable or unexpected behaviour (ie. looping).\n") % item
```